src/Home.svelte

```diff
@@ -0,0 +1,252 @@
+<script lang="ts">
+  import BasePage from "./components/BasePage.svelte";
+  let message: string = "";
+  import { parse } from "./parser";
+  import JSONTree from "svelte-json-tree";
+  import { objectView, output } from "src/store";
+  import ModeSwitcher from "./components/ModeSwitcher.svelte";
+  let selectedView: string = "content";
+  const classes = "underline justify-center border-b-4";
+
+  const decamelize = (str, separator) => {
+    separator = typeof separator === "undefined" ? "_" : separator;
+
+    return str
+      .replace(/([a-z\d])([A-Z])/g, "$1" + separator + "$2")
+      .replace(/([A-Z]+)([A-Z][a-z\d]+)/g, "$1" + separator + "$2")
+      .toLowerCase();
+  };
+
+  const hashUpdate = async () => {
+    message = decodeURIComponent(window.location.hash.substring(1));
+    parse(message);
+  };
+  const updateHash = async (message) =>
+    (window.location.hash = encodeURIComponent(message));
```

Comment:
There is a convention (started by Google) to use `#!` to indicate that the hash fragment is for application state - rather than for anchor ids that `#` is traditionally used for. More info: https://www.w3.org/blog/2011/05/hash-uris/
Here is code to add the `!`, and to allow it optionally when decoding:

```suggestion
  const hashUpdate = async () => {
    message = decodeURIComponent(window.location.hash.replace(/^#!?/, ''));
    parse(message);
  };
  const updateHash = async (message) =>
    (window.location.hash = '!' + encodeURIComponent(message));
```

Reply:
Cool, didn't knew that!